peps/pep-9999.rst

```diff
@@ -0,0 +1,145 @@
+:external+py3.14:func:`math.gcd` or :external+py3.14:func:`math.isqrt`.
+
+
+Motivation
+==========
+
+The :external+py3.14:mod:`math` documentaton says: "This module provides access
+to the mathematical functions defined by the C standard."  But as a state of
+art, over time the module was populated with functions that aren't related to
+the C standard or floating-point arithmetics.  Now it's much harder to describe
+module scope, content and interfaces (e.g. returned values).
+
+For example, the :external+py3.14:mod:`math` module documentation says: "Except
+when explicitly noted otherwise, all return values are floats."  This is not
+longer true:  *None* of the functions listed in the "Number-theoretic
+functions" [1]_ subsection of the documentation returns a float, but the
+documentation doesn't say so.  In the proposed module a similar sentence:
+"Except when explicitly noted otherwise, all return values are integers." could
+tell the truth once.
+
+Apparently, the :external+py3.14:mod:`math` can't serve as a catch-all place
+for mathematical functions: we have also the :external+py3.14:mod:`cmath` and
+the :external+py3.14:mod:`statistics`.  Let's make same for integer-related
+functions.  It would provide shared context, which reduces verbosity in the
+documentation and conceptual load.  It also aids discoverability through
+grouping related functions and IDEs suggesting helpful completions.
+
+
+Specification
+=============
+
+The PEP proposes moving the following integer-related functions [1]_ in a new
+module, called ``imath``:
+
+* :external+py3.14:func:`~math.comb`
+* :external+py3.14:func:`~math.factorial`
+* :external+py3.14:func:`~math.gcd`
+* :external+py3.14:func:`~math.isqrt`
+* :external+py3.14:func:`~math.lcm`
+* :external+py3.14:func:`~math.perm`
+
+Their aliases in :external+py3.14:mod:`math` will be :term:`soft deprecated`.
+
+Possible extensions for the new module and it's scope are discussed in the
+`Open Issues <Open Issues_>`_ section.  New functions are not part of the
+proposal.
+
+
+Backwards Compatibility
+=======================
+
+As aliases in :external+py3.14:mod:`math` will be kept for indefinite time
+(their use would be discouraged), there are no anticipated code breaks.
+
+
+Reference Implementation
+========================
+
+TBD
+
+
+Open Issues
+===========
+
+Module name
+-----------
+
+There is already an ``imath`` project on PyPI, but only with two releases, with
+the most recent one 4 years ago.  Its repository is no longer accessible.
+
+Polling shows ``intmath`` as another popular name.  The argument made was that
+the normal mathematical spelling of the imaginary unit is i, which makes imath
+ambiguous.  It also has no conflict with any PyPI module.  On the other hand,
+``intmath`` may be confused with interval math or numerical integration.
+
+Other proposed names include ``ntheory`` (like SymPy's submodule),
+``integermath`` and ``imaths``.
+
```

Comment:
You may mention that "imath" (integer) fits well with the existing "cmath" (complex) module in terms of naming consistency.